src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ClimbingPick.java

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * The {@link ClimbingPick} launches you 1 block upwards when you right click
+ * on a ice {@link Block}.
+ * Every level of efficiency {@link Enchantment} increases the launch by 0.2 blocks.
+ *
+ * @author Linox
+ *
+ */
+public class ClimbingPick extends SimpleSlimefunItem<ItemUseHandler> implements DamageableItem {
+
+    private final Map<String, Double> materialSpeedsDef = new HashMap<>();
+    private final ItemSetting<Map<String, Double>> materialSpeeds = new ItemSetting<>("launch-amounts", materialSpeedsDef);
+    private final Set<UUID> users = new HashSet<>();
+
+    public ClimbingPick(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+
+        for (Material mat : MaterialCollections.getAllIceBlocks()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+        for (Material mat : MaterialCollections.getAllConcretePowderColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+        for (Material mat : MaterialCollections.getAllTerracottaColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+
+        materialSpeedsDef.put(Material.GRAVEL.name(), 0.4);
+        materialSpeedsDef.put(Material.SAND.name(), 0.4);
+        materialSpeedsDef.put(Material.STONE.name(), 0.6D);
+        materialSpeedsDef.put(Material.DIORITE.name(), 0.6D);
+        materialSpeedsDef.put(Material.GRANITE.name(), 0.6D);
+        materialSpeedsDef.put(Material.ANDESITE.name(), 0.6D);
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)) materialSpeedsDef.put(Material.BLACKSTONE.name(), 0.6D);
+        materialSpeedsDef.put(Material.NETHERRACK.name(), 0.6D);
+
+        addItemSetting(materialSpeeds);
+    }
+
+    @Override
+    public void preRegister() {
+        super.preRegister();
+
+        addItemHandler(getBlockBreakHandler());
+    }
+
+    
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            if (e.getClickedBlock().isPresent()) {
+                Block block = e.getClickedBlock().get();
+                ItemStack item = e.getItem();
+                Player p = e.getPlayer();
+
+                if (!getID().equals("TEST_CLIMBING_PICK") && p.getLocation().distance(block.getLocation()) > 1.5) return;
+                if (e.getClickedFace() == BlockFace.DOWN || e.getClickedFace() == BlockFace.UP) return;
+
+                if (!users.contains(p.getUniqueId())) {
+                    Vector velocity = new Vector(1, 1, 1);
+                    if (!getID().equals("TEST_CLIMBING_PICK")) {
+                        velocity = p.getVelocity();
+                    }
+
+                    Material mat = block.getType();
+                    if (materialSpeeds.getValue().containsKey(mat.name())) {
+                        double launch = materialSpeeds.getValue().get(mat.name());
```

Comment:
Don't do a fucking containsKey and a get. A containsKey literally is a get anyway